cmd/secrets/read_secret.go

```diff
@@ -0,0 +1,145 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-2019 Datadog, Inc.
+
+// +build secrets
+
+package secrets
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/spf13/cobra"
+)
+
+const (
+	maxSecretFileSize    = 1024
+	compatibleMajVersion = "1"
+)
+
+type secret struct {
+	Value    string `json:"value,omitempty"`
+	ErrorMsg string `json:"error,omitempty"`
+}
+
+// ReadSecretsCmd implements a secrets backend command reading secrets from a directory/mount
+var ReadSecretCmd = &cobra.Command{
+	Use:   "read-secret",
```

Comment:
We should change the name of this command. It's misleading this we're not actually loading any secrets (secrets backend are not use, secrets cache is not updated...). What is the use case for this ? Is it related to k8s ? Should we move it to the cluster-agent code ? Why would we use this in the datadog-agent ?

Reply:
Mind clarifying why `read-secret` is a bad name? The command loads a number of secrets from files and outputs them in json. What name would you recommend to be a better fit here? This mimicks the name of the python secrets helper (`readsecret.py`).

So the point of placing this into a package that can be shared is that it presents a way of providing a default backend to load secrets in either agent or DCA binary. This is done for DCA by this PR but we can go further and replace it in the node agent as well (the python helper placed into the image under `/readsecret.py`).

Reply:
I think it's misleading since it won't execute the `secret_backend_command`. `secret` is use through all our documentation as a string that was fetch through that mechanism. Having a command called `read-secret` could fairly be interpreted as using all the mechanism in the `secret` package.

Since it's specific to k8s I think we should make that clear in the command name. What about `read-k8s-secrets` ? And updating the docstring `Read secret from a directory` to make it clear we're dealing with k8s's `secrets` feature.

WDYT ?

Reply:
We could call it `backend-read-secret` or similar?